lightning/src/ln/channel.rs

```diff
@@ -2032,6 +2061,20 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 
+		if msg.amount_msat <= (self.feerate_per_kw as u64 * HTLC_TIMEOUT_TX_WEIGHT / 1000) * 1000 + self.counterparty_dust_limit_satoshis {
+			if on_counterparty_tx_dust_inbound + on_counterparty_tx_dust_outbound + msg.amount_msat + (self.feerate_per_kw as u64 * HTLC_TIMEOUT_TX_WEIGHT / 1000) * 1000 > self.get_max_balance_dust_htlc_msat() {
+				log_info!(logger, "Cannot accept value that would put holder dusted balance {} on counterparty commitment over limit {}", on_counterparty_tx_dust_inbound + on_counterparty_tx_dust_outbound, self.get_max_balance_dust_htlc_msat());
+				pending_forward_status = create_pending_htlc_status(self, pending_forward_status, 0x1000|7);
+			}
+		}
+
+		if msg.amount_msat <= (self.feerate_per_kw as u64 * HTLC_SUCCESS_TX_WEIGHT / 1000) * 1000 + self.holder_dust_limit_satoshis {
```

Comment:
Add a lot of back-and-forth on the utility to also check the dust balance on the holder transaction as it's implicitly protected by the check on the commitment transaction, with our current assumption that holder's `dust_limit_satoshis` is *always* under counterparty's one.

Though finally I left it, as it gives us more belt-and-suspender if we touch our config settings/implementation default in the future.

Without, as of today, the worst-fee discrepancy would be 344_000 msat as explained in this Big Comment(tm): https://github.com/ariard/rust-lightning/blob/0db41f057ac4967f72646cc1e77ee59280fbf07c/lightning/src/ln/channel.rs#L2066